jgrapht-core/src/main/java/org/jgrapht/alg/AStarShortestPath.java

```diff
@@ -0,0 +1,216 @@
+            outgoingEdges=((DirectedGraph)graph).outgoingEdgesOf(currentNode.getData());
+
+
+        for(E edge : outgoingEdges){
+            V successor = Graphs.getOppositeVertex(graph, edge, currentNode.getData());
+            if (successor == currentNode.getData() || closedList.contains(successor)) //Ignore self-loops or nodes which have already been expanded
+                continue;
+
+            double gScore_current = gScoreMap.get(currentNode.getData());
+            double tentativeGScore = gScore_current + graph.getEdgeWeight(edge);
+
+            if(!vertexToHeapNodeMap.containsKey(successor) || tentativeGScore < gScoreMap.get(successor)){
+                cameFrom.put(successor, edge);
+                gScoreMap.put(successor, tentativeGScore);
+
+                double fScore= tentativeGScore + admissibleHeuristic.getCostEstimate(successor, endVertex);
+                if (!vertexToHeapNodeMap.containsKey(successor)) {
+                    FibonacciHeapNode<V> heapNode=new FibonacciHeapNode<V>(successor);
+                    openList.insert(heapNode, fScore);
+                    vertexToHeapNodeMap.put(successor, heapNode);
+                }else{
+                    openList.decreaseKey(vertexToHeapNodeMap.get(successor), fScore);
+                }
+            }
+        }
+    }
+
+    /**
+     * Builds the graph path
+     * @param startVertex starting vertex of the path
+     * @param targetVertex ending vertex of the path
+     * @param pathLength length of the path
+     * @return the shortest path from startVertex to endVertex
+     */
+    private GraphPath<V, E> buildGraphPath(V startVertex, V targetVertex, double pathLength){
+        List<E> edgeList = this.buildPath(targetVertex);
+        return new GraphPathImpl<V, E>(graph, startVertex, targetVertex, edgeList, pathLength);
+    }
+
+    /**
+     * Recursive method which traces the path from the targetVertex to the startVertex. The method traces back the path
+     * over the edges, so the method is safe to use for multi-graphs.
+     * @param currentNode node
+     * @return List of edges/arcs that constitutes the path
+     */
+    private List<E> buildPath(V currentNode){
+        if(cameFrom.containsKey(currentNode)){
+            List<E> path = buildPath(Graphs.getOppositeVertex(graph, cameFrom.get(currentNode), currentNode));
+            path.add(cameFrom.get(currentNode));
+            return path;
+        }else
+            return new ArrayList<E>();
+    }
+
+    /**
+     * Returns how many nodes have been expanded in the A* search procedure in its last invocation. A node is expanded if it is removed from the open list.
+     * @return number of expanded nodes
+     */
+    public int getNumberOfExpandedNodes(){
+        return numberOfExpandedNodes;
+    }
+
+    /**
+     * Interface for an admissible heuristic used in the A* search
+     * @param <V> vertex type
+     */
+    public interface AStarAdmissibleHeuristic<V>{
```

Comment:
You mean adding it to org.jgrapht.alg.interfaces? I hesitated to do this because it is not really an interface that will be extended with other implementations that will be included in JgraphT. Instead it is part of the AstarShortestpath algorithm implementation. Nevertheless I'm not against moving it to the interface package.